src/Client.php

```diff
@@ -140,16 +147,22 @@ public function setRawDataStream($data, $enctype = null)
      */
     public function setMethod($method = HttpRequest::METHOD_GET)
     {
-        if ($method == HttpRequest::METHOD_GET) {
-            $this->setRequestMethod(HttpRequest::METHOD_GET);
-        } elseif ($method == HttpRequest::METHOD_POST) {
-            $this->setRequestMethod(HttpRequest::METHOD_POST);
-        } elseif ($method == HttpRequest::METHOD_PUT) {
-            $this->setRequestMethod(HttpRequest::METHOD_PUT);
-        } elseif ($method == HttpRequest::METHOD_DELETE) {
-            $this->setRequestMethod(HttpRequest::METHOD_DELETE);
-        } elseif ($method == HttpRequest::METHOD_HEAD) {
-            $this->setRequestMethod(HttpRequest::METHOD_HEAD);
+        switch (strtoupper($method)) {
```

Comment:
original functionality doesn't upper passed method, why it is converted ?

Reply:
Because of change to strict equality check, which was then undone by change to `switch` which is not strict!

Reply:
I prefer to not change when possible to avoid regression, exclude on cs if required.